### ISSUE TITLE
applyPatch affecting wrong line number with with new lines

### ISSUE THREAD

**Issue by Luiz-N** [kpdecker/jsdiff#159]
@kpdecker your library is great so far. I think I might have found an edge case unless i'm missing something?
```
oldString = 'test\ntest2\n \ntest3\n';
"test
test2
 
test3
"
```
```
newString = 'test\ntest2\ntest3\n';
"test
test2
test3
"
```
Now I create a patch
```js
d = JsDiff.createPatch('file/path.js', oldString, newString, '', '', { context: 0 });
```
This patch/diff looks correct
```diff
"Index: file/path.js
===================================================================
--- file/path.js	
+++ file/path.js	
@@ -3,1 +3,0 @@
- 
```
However when I apply the patch....
```
result = JsDiff.applyPatch(oldString, d);
"test
test2
 
"
```
We can see it seems to be affecting the wrong line number. Is it trimming somewhere on the oldString where it shouldn't be?

**Comment by Luiz-N** [comment]
**UPDATE:**

I've narrowed the cause of the bug down to when context is set to 0. So by setting it to 1 or higher it works correctly:
```
d = JsDiff.createPatch('file/path.js', oldString, newString, '', '', { context: 1 });

result = JsDiff.applyPatch(oldString, d);
"test
test2
test3
"
```

**Comment by ibc** [comment]
Good catch. Same happens here.

**Comment by bp-dev** [comment]
Also having problems with empty lines: a patch starting with a empty line of context such as
```
@ -93,6 +93,19 @@

         <div id="secondaryToolbar" class="secondaryToolbar hidden doorHangerRight">
           <div id="secondaryToolbarButtonContainer">
+            <button id="secondaryOnePageMode" class="secondaryToolbarButton onePageMode visibleLargeView" title="One Page Mode" tabindex="51" data-l10n-id="one_page_mode">
```
is not applied (and following blocks are applied in the wrong place as line numbers don't match anymore)

Removing the empty line:
```
@@ -93,5 +93,18 @@
         <div id="secondaryToolbar" class="secondaryToolbar hidden doorHangerRight">
           <div id="secondaryToolbarButtonContainer">
+            <button id="secondaryOnePageMode" class="secondaryToolbarButton onePageMode visibleLargeView" title="One Page Mode" tabindex="51" data-l10n-id="one_page_mode">
```
makes it work fine.

**Comment by Luiz-N** [comment]
I think this got fixed with #212
